sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -37,6 +37,20 @@ import org.apache.spark.util.{MutableURLClassLoader, Utils}
  */
 private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
+  // System preserved database should not exists in metastore. However it's hard to guarantee it
+  // for every session, because case-sensitivity differs. Here we always lowercase it to make our
+  // life easier.
+  val globalTempDB = sparkContext.conf.get(GLOBAL_TEMP_DATABASE).toLowerCase
```

Comment:
btw, do we check if there is an existing db having the same name as `GLOBAL_TEMP_DATABASE`?